goldstone/core/tasks.py

```diff
@@ -82,15 +89,17 @@ def update_persistent_graph():
        - updated from the cloud if they are already in the graph.
 
     """
-    from goldstone.cinder.utils import update_nodes as update_cinder_nodes
-    from goldstone.glance.utils import update_nodes as update_glance_nodes
-    from goldstone.keystone.utils import update_nodes as update_keystone_nodes
-    from goldstone.nova.utils import update_nodes as update_nova_nodes
-
-    update_cinder_nodes()
-    update_glance_nodes()
-    update_keystone_nodes()
-    update_nova_nodes()
+
+    graph_resources = [cinder_res, glance_res, keystone_res, nova_res]
+
+    for obj in graph_resources:
+        try:
+            cinder_res.update_nodes()
+            glance_res.update_nodes()
+            keystone_res.update_nodes()
+            nova_res.update_nodes()
+        except Exception as e:
+            logger.critical(str(e))
```

Comment:
logger.exception throws an exception which halts the celery task. Makes the try-catch block a moot. Hence logger.critical()

...2016-02-10T00:22:09+0000 ERROR    goldstone.core.tasks 
Traceback (most recent call last):
  File "/home/app/goldstone/core/tasks.py", line 94, in update_persistent_graph
    glance_res.update_nodes()
  File "/venv/local/lib/python2.7/site-packages/mock.py", line 955, in __call__
    return _mock_self._mock_call(*args, **kwargs)
  File "/venv/local/lib/python2.7/site-packages/mock.py", line 1015, in _mock_call
    raise result
Exception
2016-02-10T00:22:09+0000 ERROR    goldstone.core.tasks 
Traceback (most recent call last):
  File "/home/app/goldstone/core/tasks.py", line 94, in update_persistent_graph
    glance_res.update_nodes()
  File "/venv/local/lib/python2.7/site-packages/mock.py", line 955, in __call__
    return _mock_self._mock_call(*args, **kwargs)
  File "/venv/local/lib/python2.7/site-packages/mock.py", line 1013, in _mock_call
    result = next(effect)
StopIteration
2016-02-10T00:22:09+0000 ERROR    goldstone.core.tasks 
Traceback (most recent call last):
  File "/home/app/goldstone/core/tasks.py", line 94, in update_persistent_graph
    glance_res.update_nodes()
  File "/venv/local/lib/python2.7/site-packages/mock.py", line 955, in __call__
    return _mock_self._mock_call(*args, **kwargs)
  File "/venv/local/lib/python2.7/site-packages/mock.py", line 1013, in _mock_call
    result = next(effect)
StopIteration